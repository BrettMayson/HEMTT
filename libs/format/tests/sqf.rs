use hemtt_format::{FormatterConfig, format_sqf};

const ROOT: &str = "tests/sqf/";

macro_rules! sqf {
    ($dir:ident) => {
        paste::paste! {
            #[test]
            fn [<sqf_ $dir>]() {
                insta::assert_snapshot!(sqf(stringify!($dir)));
            }
        }
    };
}

sqf!(arrays);
sqf!(blocks);
sqf!(cba);
sqf!(comments);
sqf!(format);
sqf!(hash);
sqf!(if_and);
sqf!(if_not);
sqf!(if);
sqf!(include);
sqf!(macro_path);
sqf!(macros);
sqf!(numbers);
sqf!(preserve_lines);
sqf!(private);
sqf!(sameline);

fn sqf(file: &str) -> String {
    format_sqf(
        &std::fs::read_to_string(format!("{ROOT}{file}.sqf")).expect("Failed to read test file"),
        &FormatterConfig::default(),
    )
    .unwrap_or_else(|err| format!("ERROR: {err}"))
}
