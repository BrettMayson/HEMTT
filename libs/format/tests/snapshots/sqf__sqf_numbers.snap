---
source: libs/format/tests/sqf.rs
expression: sqf(stringify! (numbers))
---
private _number = 0xDEADBEEF;
private _number = 0x123;
private _exponent = 123e10;
private _exponent = 123e-10;
private _int = 123;
private _float = 123.456;
private _negative = -123;
private _negativeFloat = -123.456;
private _complex = -123.456e-10;

private _addition = 1 + 2;
private _subtraction = 1 - 2;
private _negativeSubtraction = -1 - -2;

private _brackets = (1 + 2) * ( -1 - -2);
